content/en/blog/2021/stateful-set-made-easier/index.md

```diff
@@ -0,0 +1,226 @@
+---
+title: "StatefulSets Made Easier With Istio 1.10"
+description: Learn how to easily deploy StatefulSets with Istio 1.10.
+publishdate: 2021-05-18
+attribution: "Lin Sun (Solo.io), Christian Posta (Solo.io), John Howard (Google), Zhonghu Xu (Huawei)"
+keywords: [statefulset,Istio,networking,localhost,loopback,eth0]
+---
+
+Kubernetes [`StatefulSets`](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/) are commonly used to manage stateful applications. In addition to managing the deployment and scaling of a set of `Pods`, `StatefulSets` provide guarantees about the ordering and uniqueness of those `Pods`. Common applications used with StatefulSets include ZooKeeper, Cassandra, Elasticsearch, Redis and NiFi.
+
+The Istio community has been making gradual progress towards zero-configuration support for `StatefulSets`; from automatic mTLS, to eliminating the need to create `DestinationRule` or `ServiceEntry` resources, to the most recent [pod networking changes in Istio 1.10](/blog/2021/upcoming-networking-changes/).
+
+What is unique about using a `StatefulSet` with a service mesh? The `StatefulSet` pods are created from the same spec, but are not interchangeable: each has a persistent identifier that it maintains across any rescheduling. It is common for pods within a `StatefulSet` to listen on the pod IP only, instead of `0.0.0.0`, for communication within the `StatefulSet` pods.
+
+Many [issues](https://github.com/istio/istio/issues/10659) have been reported from the Istio community around `StatefulSets`.
+
+Using ZooKeeper as an example, from the `zoo.cfg` file in our ZooKeeper pod, it is configured not to listen on all IPs for quorum communication within the `StatefulSet` pods.
+
+{{< text plain >}}
+quorumListenOnAllIPs=false
+{{< /text >}}
```

Comment:
```suggestion
What is unique about using a `StatefulSet` with a service mesh? The `StatefulSet` pods are created from the same spec, but are not interchangeable: each has a persistent identifier that it maintains across any rescheduling. The kind of apps that run in a `StatefulSet` are often those that need to communicate amongst their pods, and, as they come from a world of hard-coded IP addresses, may listen on the pod IP only, instead of `0.0.0.0`.

ZooKeeper, for example, is configured by default not to listen on all IPs for quorum communication:

{{< text plain >}}
quorumListenOnAllIPs=false
{{< /text >}}

Over the last few releases, the Istio community has [reported many issues](https://github.com/istio/istio/issues/10659) around support for applications running in `StatefulSets`.
```

Thoughts on that? @linsun @howardjohn

Reply:
Only thing I'm not sure is `and, as they come from a world of hard-coded IP addresses,` otherwise LGTM

@howardjohn ?

Reply:
Sounds good enough to me, but I am not familiar with non-k8s world :slightly_smiling_face:

Reply:
Thanks Craig, we'll take your suggestions!